clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3454,6 +3450,153 @@ static void handleWorkGroupSizeHint(Sema &S, Decl *D, const ParsedAttr &AL) {
                              AL.getArgAsExpr(2));
 }
 
+// Handles max_work_group_size attribute.
+// If the declaration has a SYCLIntelMaxWorkGroupSizeAttr,
+// check to see if the attribute holds equal values (1, 1, 1).
+// Returns true if the attribute values are not equal to one.
+static bool InvalidWorkGroupSizeAttrs(const Expr *E, const Expr *E1,
+                                      const Expr *E2, const Expr *E3) {
+  // If any of the operand is still value dependent, we can't test anything.
+  const auto *CE = dyn_cast<ConstantExpr>(E);
+  const auto *CE1 = dyn_cast<ConstantExpr>(E1);
+  const auto *CE2 = dyn_cast<ConstantExpr>(E2);
+  const auto *CE3 = dyn_cast<ConstantExpr>(E3);
+
+  if (!CE || !CE1 || !CE2 || !CE3)
+    return false;
+
+  // Otherwise, check if the attribute values are equal to one.
+  if (CE->getResultAsAPSInt() == 0 &&
+      (CE1->getResultAsAPSInt() != 1 || CE2->getResultAsAPSInt() != 1 ||
+       CE3->getResultAsAPSInt() != 1)) {
+    return true;
+  }
+  return false;
```

Comment:
Just need one `return` statement here.
```
return CE->getResultAsAPSInt() == 0 &&
           CE1->getResultAsAPSInt() == 1 && CE2->getResultAsAPSInt() == 1 &&
           CE3->getResultAsAPSInt() == 1;
```

Reply:
Added one return statement. The patch gives error if function returns true, so logic should be here:
```
return (CE->getResultAsAPSInt() == 0 &&
            (CE1->getResultAsAPSInt() != 1 && CE2->getResultAsAPSInt() != 1 &&
            CE3->getResultAsAPSInt() != 1));
```